modules/oauth/src/main/java/org/mule/extension/oauth2/internal/ParameterExtractor.java

```diff
@@ -32,7 +37,7 @@ public String getParamName() {
   /**
    * @return value extracted from the token response.
    */
-  public String getValue() {
+  public Function<Event, String> getValue() {
```

Comment:
Those would be refactored out with MULE-11305. It will need the new muleClient though.